src/components/Asset/AssetActions/Download.tsx

```diff
@@ -71,23 +71,29 @@ export default function Download({
         tokenInLiquidity: poolData?.baseTokenLiquidity,
         tokenOutLiquidity: poolData?.datatokenLiquidity,
         tokenOutAmount: '1',
-        opcFee: getOpcFeeForToken(poolData.baseToken.address, asset?.chainId),
+        opcFee: getOpcFeeForToken(
+          poolData?.baseToken?.address || asset?.accessDetails?.addressOrId,
+          asset?.chainId
+        ),
         lpSwapFee: poolData?.liquidityProviderSwapFee,
         publishMarketSwapFee: poolData?.publishMarketSwapFee,
         consumeMarketSwapFee: '0'
       }
+
       const orderPriceAndFees = await getOrderPriceAndFees(
         asset,
         ZERO_ADDRESS,
         params
       )
 
       setOrderPriceAndFees(orderPriceAndFees)
-      setIsDisabled(false)
+      if (isAssetNetwork) {
```

Comment:
in theory, this makes `isAssetNetwork` a dependency of the effect, just like the removed `isLoading` so should be added to dependency array. That is unless something else in existing dependency would change on wallet network change too, in which case we would know better than ESLint and we should add an ESLint-disabling comment on the dependency array so the warning here doesn't appear:

<img width="751" alt="Screen Shot 2022-04-27 at 21 48 29" src="https://user-images.githubusercontent.com/90316/165618445-946ed45c-fe8a-498b-a342-03b6d1932b9a.png">

Reply:
another thing, we might not need to call `setIsDisabled()` here based on `isAssetNetwork` as further down in the next effect we only deal with the disable state of the button.

Reply:
removed both as they're not necessary